### ISSUE TITLE
Support Amazon Q as an Inline Assistant

### ISSUE THREAD

**Issue by 5herlocked** [zed-industries/zed#18274]
### Check for existing issues

- [X] Completed

### Describe the feature

Add [Amazon Q](https://aws.amazon.com/q/developer/) as an Inline Assistant provider.

**Comment by notpeter** [comment]
Some links:
- [Amazon BedRock API reference](https://docs.aws.amazon.com/bedrock/latest/APIReference/API_Operations_Amazon_Bedrock.html) (Amazon Q is built on BedRock)
- [amazon-q-vscode vscode extension source](https://github.com/aws/aws-toolkit-vscode/tree/master/packages/amazonq)
- [aws-sdk-rust bedrock-runtime example](https://github.com/awslabs/aws-sdk-rust/tree/main/examples/examples/bedrock-runtime)

Looks like they support inline suggestions (ala Copilot) and Chat, plus some other agent-based things.

**Comment by Xenoha** [comment]
Bump!

This would complete the editor for me. I'm sure this is just a when, but I hope the when is coming really soon.

**Comment by soundstep** [comment]
Bump 2025, can I do that?!

**Comment by soundstep** [comment]
Some progress here actually, if it is relevant:
https://github.com/zed-industries/zed/issues/16544#issuecomment-2686327916

Does this mean we can use Amazon Q already somehow? With vscode it is kinda linked with [AWS Builder ID](https://docs.aws.amazon.com/signin/latest/userguide/sign-in-aws_builder_id.html).

**Comment by sFritsch09** [comment]
Would be nice to support CLI in terminal: https://github.com/aws/amazon-q-developer-cli

**Comment by zalez** [comment]
> Would be nice to support CLI in terminal: https://github.com/aws/amazon-q-developer-cli

Yes, this works already. The terminal inside Zed is "just a a terminal", and as such, it is capable of running the Q Developer CLI just fine. I’ve done it a few time and can confirm it works. It's just limited to what Q Developer can do in the terminal, i. e. it doesn't do any of the editor in-line code completions or /dev chat modes it would on other IDEs. That requires extra work.

But as an interim solution, Q Developer CLI inside a terminal inside Zed works well.

**Comment by sFritsch09** [comment]
Yeah, you are right I was actually referring to autocompletion :)

**Comment by alioshr** [comment]
Autocompletion does not work on Zed's terminal... Probably lacks an integration port on Amazon's side?

**Comment by inluxc** [comment]
Do we have any news for this?

**Comment by demo-err** [comment]
there is an aws [subreddit](https://www.reddit.com/r/aws/). i think that it is moderated by actual aws employees. maybe we should request the q-zed integration there.

**Comment by saurabh-hirani** [comment]
cli support doesn't require changes to the editor but having a full fledged support the way vscode does it would be really useful - https://marketplace.visualstudio.com/items?itemName=AmazonWebServices.amazon-q-vscode

**Comment by sFritsch09** [comment]
> Autocompletion does not work on Zed's terminal... Probably lacks an integration port on Amazon's side?

It is like @alioshr said, you need to configure the connection to the CLI, Extensions are integrated here: 
https://github.com/aws/amazon-q-developer-cli/tree/main/extensions

**Comment by jrussellfreelance** [comment]
Would love full support as well like i can configure openrouter and openai.  CLI is subpar option at this point.

**Comment by rkjnsn** [comment]
It looks like Amazon Q Developer CLI has been superseded by the closed-source Kiro CLI, which now has ACP support built in: https://kiro.dev/docs/cli/acp/

It does not appear that history is yet supported.
